### PR TITLE
[ENH] use logger instead of print statements

### LINKED ISSUE #62: use a logger instead of print statement
allows
- different levels of verbosity
- having log files that can help with helping users

### PR DESCRIPTION
**Pull request by Remi-Gau**
- closes #62 
---

- add logger
- set verbosity level from CLI
- remove print statement and passing verbosity level as argument

### CODE REVIEW

File: giga_connectome/mask.py [hunk 1]
```diff
@@ -244,12 +239,11 @@ def _check_mask_affine(
     common_affine = max(
         set(header_info["affine"]), key=header_info["affine"].count
     )
-    if verbose > 0:
-        print(
-            f"We found {len(set(header_info['affine']))} unique affine "
-            f"matrices. The most common one is "
-            f"{key_to_header[common_affine]}"
-        )
+    gc_log.debug(
```
Comment (Remi-Gau) [inline giga_connectome/mask.py:242]:
```suggestion
    gc_log.info(
```

File: giga_connectome/utils.py [hunk 2]
```diff
@@ -196,6 +200,7 @@ def check_path(path: Path, verbose=True):
         ]
         n = str(max(existing_numbers) + 1) if existing_numbers else "1"
         path = path_parent / f"{path.stem}_{n}{ext}"
-        if verbose:
-            print(f"Specified path already exists, using {path} instead.")
+
+        gc_log.debug(f"Specified path already exists, using {path} instead.")
```
Comment (Remi-Gau) [inline giga_connectome/utils.py:204]: not sure if this one should be info or debug?

**Comment by Remi-Gau** [comment]
have kind of assumed that:
- versbosity > 0 --> info
- versbosity > 1 --> debug

**Comment by Remi-Gau** [comment]
Actually added rich formating:

Would look like this:

image

**Comment by htwangtw** [comment]
> Let me know if you want to use Rich to get prettier log outputs:
> 
> https://rich.readthedocs.io/en/stable/logging.html

yes.

**Comment by htwangtw** [comment]
I have played with it through some deep learning experiment framework. it's nice

File: giga_connectome/utils.py (hunk 2 above)
Comment (htwangtw) [reply giga_connectome/utils.py:204]: This is related to #76....
I am inclined to call this info for what we have for now.
But after #76 this should not be relevant?

**Review by htwangtw — APPROVED** [review]
All things fixed - good to go

**Pull request merged**